### DISCUSSION TITLE
Multiple predictor variables (still) supported for mcp models?

### DISCUSSION THREAD

**Discussion by mws19** [lindeloev/mcp#158] [Q&A] [answered]
Hello - I'm writing to ask whether mcp can be used to construct a segmented linear regression model with multiple predictor variables (specifically, one continuous and several categorical). A google search of this very question led me to [this discussion thread](https://github.com/lindeloev/mcp/issues/90) regarding development of v0.4. I thought that by downloading this version as described in the discussion my problems would be solved, but I'm still unable to include multiple predictor variables in my model, getting an error that the model `Found more than one x-variable in segment 1`. When I include only a single continuous variable, the model fits well. I've also tried downloading the latest development version from GitHub, and encounter the same problem. 

With both package versions, the functionality with dealing with multiple predictor variables seems to be absent altogether, because I also can't run the example model for multiple predictors  `result = mcp_example("multiple")` - I get the following: `Error in assert_value(name, c("demo", "ar", "binomial", "intercepts",  : name must be one of 'demo', 'ar', 'binomial', 'intercepts', 'rel_prior', 'quadratic', 'trigonometric', 'varying', 'variance'. Got multiple`

Am I missing something, or has the mcp functionality for modeling multiple predictor variables indeed been discontinued? If so, any recommendations for an alternative method to fit such a model? I suppose I could use mcp to find the changepoint in my data, subset accordingly, and then fit two separate models in brms? Thanks in advance for any insight!

**Comment by lindeloev** [comment, marked as answer]
mcp v0.4 has not been released yet, unfortunately. Only a few edge case bugs remain to be ironed out and it will be awesome! But I prioritized working on another project (https://giveffektivt.dk - an investor-like "maximum value-on-investment" approach to finding the best/most-effective charities). I really want to get back and finish it, but I can't give a timeline when I'll finish it up. It's in [the v0.4 branch](https://github.com/lindeloev/mcp/tree/v0.4) but it may break due to some backwards-incompatible changes in dependency-packages.

**Reply by mws19** [reply]
Circling back around to this, as I take up the relevant analysis again - just want to say thanks for your reply, and I look forward to checking out mcp v0.4 whenever it's released!
